File: PKI/src/main/java/rs/tim33/PKI/DTO/Certificate/CreateCertificateDTO.java
package rs.tim33.PKI.DTO.Certificate;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;

import rs.tim33.PKI.Models.CertificateType;

public class CreateCertificateDTO {
	//The certificate data
	public CertificateType certType;
	public Long issuerId;
	public SubjectDTO subject;
	public Long templateId = 0L;
	
	public LocalDateTime notBefore;
	public LocalDateTime notAfter;
	
	//EXTENSIONS
	//SAN
	public List<TypeValue> san;
	
	//KeyUsage
	public List<String> keyUsage;
	
	//ExtendedKeyUsage
	public List<String> extendedKeyUsage;
	
	//BasicConstraints
	public Integer pathLenConstraint;
}

File: FrontPKI/FrontPKI/src/app/DTO/Certificate/CreateCertificateDTO.ts
import { SubjectDTO } from "./SubjectDTO";
import { TypeValue } from "./TypeValue";

export class CreateCertificateDTO{
	//The certificate data
	issuerId: number = 0;
	subject: SubjectDTO = new SubjectDTO();
	notBefore: String = "";
	notAfter: String = "";
	
	//EXTENSIONS
	//SAN
	san: TypeValue[] = [];
	
	//KeyUsage
	keyUsage: String[] = [];
	
	//ExtendedKeyUsage
	extendedKeyUsage: String[] = [];
	
	//BasicConstraints
	isEndEntity: Boolean = false;
	pathLenConstraint: number = -1;
}